### ISSUE TITLE
Alignment with Truck GT in TnT supplied dataset

### ISSUE THREAD

**Issue by yanivw12** [hbb1/2d-gaussian-splatting#111]
Hi,

I downloaded the pre-processed TnT dataset that you supplied, and ran into an issue:
If I take the sparse COLMAP points and apply the transformation from the Tanks and Temples official site (*_trans.txt), the transformed points should align with the GT point cloud. This is the case for Barn for example:

![image](https://github.com/hbb1/2d-gaussian-splatting/assets/60115056/b5e51588-2fa1-4e11-8727-1a99a99198d6)

This also works for the other scans. However, for Truck, and only for Truck, they do not align:

![image](https://github.com/hbb1/2d-gaussian-splatting/assets/60115056/c19a9ec8-1299-438a-9699-30cb4ea04e68)

I'm taking TNT_GOF/TrainingSet/Truck/sparse/0/points3D.ply as the sparse COLMAP points, and using the Truck_trans.txt and Truck.ply files from the official TnT site.

I'd appreciate your help in identifying the reason for this - am I missing something?

Thanks in advance.

**Comment by hbb1** [comment]
Hi, does the truck data work normally by running the evaluation?

**Comment by yanivw12** [comment]
I'm running the official TnT evaluation code on a mesh that aligns perfectly with the above points3D file, and I input the Truck_COLMAP_SfM.log file that you supplied. The result is a low score because the mesh and GT aren't aligned. This is the precision.ply I get from the TnT evaluation:

![image](https://github.com/hbb1/2d-gaussian-splatting/assets/60115056/5b5ba627-4f97-4585-a419-4832549df159)

As you can see, because the transformed mesh and GT do not align, the crop is incorrect and therefore the metrics are very low.

**Comment by yanivw12** [comment]
Sorry, clicked on close issue by mistake

**Comment by hbb1** [comment]
I think you input the incorrect transformation file.

**Comment by yanivw12** [comment]
I triple-checked that it is the correct file. This is the transformation I'm using:

![image](https://github.com/hbb1/2d-gaussian-splatting/assets/60115056/9174f778-c9e1-49db-887c-e9c19ca58fbd)

It matches the transformation from the TnT site:

1.138071816345332055e+00 -7.300598829132702583e-02 6.497527927834608752e-01 -2.323759808085589018e+00
-6.536992969315738033e-01 -9.985495936658358995e-02 1.133764632318774668e+00 1.657361383604702088e+00
-1.363067303782293342e-02 -1.306680962583968819e+00 -1.229434503706130222e-01 -7.163783983342566941e+00
0.000000000000000000e+00 0.000000000000000000e+00 0.000000000000000000e+00 1.000000000000000000e+00

**Comment by yanivw12** [comment]
This is the transformation I'm using:

![image](https://github.com/hbb1/2d-gaussian-splatting/assets/60115056/e391f4a9-2dd7-415e-a7ef-9780ae387de5)

It matches with the transformation from the TnT site:

1.138071816345332055e+00 -7.300598829132702583e-02 6.497527927834608752e-01 -2.323759808085589018e+00
-6.536992969315738033e-01 -9.985495936658358995e-02 1.133764632318774668e+00 1.657361383604702088e+00
-1.363067303782293342e-02 -1.306680962583968819e+00 -1.229434503706130222e-01 -7.163783983342566941e+00
0.000000000000000000e+00 0.000000000000000000e+00 0.000000000000000000e+00 1.000000000000000000e+00

When I run the same code for other scans, I get a perfect alignment.

**Comment by hbb1** [comment]
Did the evaluation for other scans work normally?

**Comment by yanivw12** [comment]
Yes. For example, precision.ply on Ignatius:

![image](https://github.com/hbb1/2d-gaussian-splatting/assets/60115056/26e65a25-90d5-4b4f-a5b1-0e141b610bac)

**Comment by hbb1** [comment]
How about the quantitative results?

**Comment by yanivw12** [comment]
I get this, which is obviously lower than it should be:

![image](https://github.com/user-attachments/assets/3eb20838-5b6f-4280-a320-a4eacf13b9e2)

And again, the reconstructed mesh from 2dgs (yellow) after transformation+crop and the gt mesh don't align:

![image](https://github.com/user-attachments/assets/bf6fbf5a-8934-4538-8d4b-cdc3a3c5d512)

I noticed that the COLMAP poses for Truck that you supplied are quite different from the SfM log on the TnT website, or even from the COLMAP poses I got when running it directly on the source images:

![image](https://github.com/user-attachments/assets/62957f7e-1915-40fd-9b19-db8605796eb8)

Left is your supplied SfM, right is from the TnT website.

I'd appreciate if you could share how you pre-processed the images to get the COLMAP poses - that'll help me debug the issue.

**Comment by hbb1** [comment]
Hi. I want to know, is the quantitative result look normally by running our provided script? Does it also face the same issue? Due to the storage limit, we use the data preprocessed and shared by [GOF](https://github.com/autonomousvision/gaussian-opacity-fields), which you can refer to. But, early, we have processed the data by ourselves, by running a script [here](https://github.com/hbb1/colmap_runner), which should also work normally.

**Comment by yanivw12** [comment]
The results I showed in my previous message were from running your provided script. The issue is with the original COLMAP poses, something seems to be wrong with them, and I'm still trying to figure out what. Also, when you evaluated it for your paper, you got better results - did you use the same COLMAP poses? do you have the precision and recall ply files to check if they're aligned?

**Comment by hbb1** [comment]
Hi, I use the data and evaluation script to reproduce the results. I have a double check but found there is no issue. Perhaps there is an issue with your downloaded files or running commands.

**Comment by hbb1** [comment]
Do other scans, except the Truck, work normally? I need more details to figure out the issue.

**Comment by hbb1** [comment]
The only reason I can think of is that you supplied a wrong COLMAP_SfM.log file. The file should has been included in `TNT_GOF/TrainingSet/{scene}`

**Comment by yanivw12** [comment]
I double checked, and I'm supplying the correct SfM file. I downloaded the TNT_GOF dataset again and got the same file:
![image](https://github.com/user-attachments/assets/c3a2944d-e0b6-45fd-a638-9ae049d6ebd0)
Can you confirm that this is the same SfM file you're using?

**Comment by hbb1** [comment]
> Do other scans, except the Truck, work normally? I need more details to figure out the issue.

I cannot access to my server right now. BTW, can you answer this question?

**Comment by yanivw12** [comment]
Yes, for example, Ignatius scan works perfectly fine and I get the expected results:
![image](https://github.com/user-attachments/assets/a5ba378e-c5ef-4246-8a67-fcb28855501b)
(Yellow and grey meshes are aligned)
![image](https://github.com/user-attachments/assets/b282f233-6114-41e1-b1f6-31c2a62b1c75)

**Comment by hbb1** [comment]
Okay, then the Truck data must have an issue with its COLMAP_SfM.log file. I am not sure if the data is modified.
If so, there should be issue with GOF https://github.com/autonomousvision/gaussian-opacity-fields/blob/main/scripts/run_tnt.py#L39

Because our server has been broken down, I cannot have an immediate check. If you need quick feedback, I suggest opening issue under GOF's repo.

**Comment by yanivw12** [comment]
Ok, thanks, I'll open an issue there.

**Comment by ThePassedWind** [comment]
> Ok, thanks, I'll open an issue there.

@yanivw12 
Hi, have you solved this problem?
